modules/bpi/bpi.images.inc

```diff
@@ -88,32 +91,45 @@ function bpi_assets_form($form, $form_state, $assets = array()) {
  * @ingroup forms
  */
 function bpi_assets_form_submit($form, &$form_state) {
-  $input = array_keys($form_state['input']);
+  $bpi_assets = $_SESSION['bpi']['assets'];
+
```

Comment:
Why is session handling necessary? Could we store the assets as a form value instead?